src/main/java/com/puppycrawl/tools/checkstyle/AstTreeStringPrinter.java

```diff
@@ -114,10 +117,13 @@ private static String printJavaAndJavadocTree(DetailAST ast) {
      * @return string javadoc tree
      */
     private static String parseAndPrintJavadocTree(DetailAST node) {
-        final DetailNode tree = DetailNodeTreeStringPrinter.parseJavadocAsDetailNode(node);
+        final DetailAST javadocBlock = node.getParent();
+        final DetailNode tree = DetailNodeTreeStringPrinter.parseJavadocAsDetailNode(javadocBlock);
 
-        final String rootPrefix = getIndentation(node);
-        final String prefix = rootPrefix.substring(0, rootPrefix.length() - 2) + "   ";
+        String baseIdentation = getIndentation(node);
+        baseIdentation = baseIdentation.substring(0, baseIdentation.length() - 2);
+        final String rootPrefix = baseIdentation + "   `--";
+        final String prefix = baseIdentation + "       ";
```

Comment:
I do not like this hardcode of indentation for `prefix` and `baseIndentation`. Can you please make one more input file with javadocs on methods.